### DISCUSSION TITLE
Where Mandarine is heading

### DISCUSSION THREAD

**Discussion by andreespirela** [nestdotland/nest.land#218] [Show and tell]
Many of you may already have heard of [MandarineTS](https://github.com/mandarineorg/mandarinets), a MVC decorator-driven framework to create complex applications with design patterns like SOLID, Dependency Injection, and other practices [(Click here to see documentation)](https://mandarineframework.gitbook.io/mandarine-ts/).

Mandarine is now getting bigger as well as the Deno community, and I am aiming for Mandarine to become the official MVC framework for Deno (at a community level)

![Mandarine.ts Logo](https://user-images.githubusercontent.com/60560109/81625975-6c6c4a80-93c8-11ea-8cd3-c5718fd56d52.png)

Right now, the future is really uncertain for both Deno & Mandarine. Mandarine is in its early versions as well as Deno, and many of the changes that Deno is bringing can turn out to be breaking changes, thus putting _Mandarine's stability_ at **risk**.

------------------------

# Production

The main goal of **Mandarine** right now is to be able to fully work in a production environment. Take a look [at this issue](https://github.com/mandarineorg/mandarinets/issues/57).
In order for Mandarine to be able to be production-ready, there are many steps we need to go through such as adding benchmarks, a lot of code refactoring, improve functionalities, and many others. Mandarine has about **13k LOC** which means every single line of code needs to be _thought carefully_ . But I would suggest, the **most important step** to be _production-ready- is to find a community not only of _collaborators_ but of users. Mandarine needs people using it in order to grow, this is just like a _domino effect_, the more people using Mandarine, the more bugs and feature requests we will get which would directly impact Mandarine's growth rate in terms of usability.

# Thoughts?
 - Have you ever considered joining a open-source project like Mandarine?
 - Have you ever used or want to use Mandarine?
 - Any other thought you have on this or on what is being said on this discussion?

Let me know your thoughts on the discussion below!

**Comment by t8** [comment]
This is awesome! Thanks for sharing Mandarine. I think that people will find the Mandarine project to be a great place for contributing and building their Deno applications.

**Comment by martonlederer** [comment]
I haven't tried mandarine, but I'm planning to. Keep up the good work
